From: "danpetry" <dpetry@anaconda.com>
Date: Thu, 6 Mar 2025
Subject: Point lib paths to $PREFIX/lib

Point cpp_extension, which is used in cpp_builder to compile at runtime,
towards the lib directory rather than the site-packages directory, because
that's where are sos/dlls are.

Index: pytorch/torch/utils/cpp_extension.py
===================================================================
diff --git torch/utils/cpp_extension.py torch/utils/cpp_extension.py
index a8caba3..b2d28ab 100644
--- a/torch/utils/cpp_extension.py
+++ b/torch/utils/cpp_extension.py
@@ -39,7 +39,18 @@ SHARED_FLAG = '/DLL' if IS_WINDOWS else '-shared'
 
 _HERE = os.path.abspath(__file__)
 _TORCH_PATH = os.path.dirname(os.path.dirname(_HERE))
-TORCH_LIB_PATH = os.path.join(_TORCH_PATH, 'lib')
+if os.environ.get("CONDA_BUILD", None) is not None:
+    # regular build (& testing) phase --> PREFIX is set
+    # linux: PREFIX + lib
+    # windows: PREFIX + Library/bin/
+    pieces = [os.environ["PREFIX"]] + ["Library/bin" if IS_WINDOWS else "lib"]
+    TORCH_LIB_PATH = os.path.join(*pieces)
+elif os.environ.get("CONDA_PREFIX", None) is not None:
+    # final environment
+    pieces = [os.environ["CONDA_PREFIX"]] + ["Library/bin" if IS_WINDOWS else "lib"]
+    TORCH_LIB_PATH = os.path.join(*pieces)
+else:
+    TORCH_LIB_PATH = os.path.join(_TORCH_PATH, 'lib')
 
 
 SUBPROCESS_DECODE_ARGS = ('oem',) if IS_WINDOWS else ()
